waitgroup.go

```diff
@@ -0,0 +1,67 @@
+package udp
+
+import (
+	"sync"
+)
+
+// A WaitGroup waits for a collection of goroutines to finish.
+// The main goroutine calls Add to set the number of
+// goroutines to wait for. Then each of the goroutines
+// runs and calls Done when finished. At the same time,
+// Wait can be used to block until all goroutines have finished.
+//
+// WaitGroups in the sync package do not allow adding or
+// subtracting from the counter while another goroutine is
+// waiting, while this one does.
+//
+// A WaitGroup must not be copied after first use.
+//
+// In the terminology of the Go memory model, a call to Done
+type WaitGroup struct {
```

Comment:
As it's not really related to `udp` module, it would be better to make a submodule like `./pkg/sync` if exposing this

Reply:
Sure. I created `./pkg/sync` package .